haproxy_agent.py

```diff
@@ -27,6 +28,10 @@
 
 SPOA_AGENT = SpoaServer()
 
+# todo: better error checking
+assert SHARED_KEY != "", "NC_HARP_SHARED_KEY is not set"
```

Comment:
just fyi, this is checked here:

https://github.com/cloud-py-api/HaRP/blob/9a46744b86c302d942623212562b9662bd864df5/start.sh#L147-L172